src/test/java/org/biouno/unochoice/issue65235/TestProjectObjectIsPresentOnMultibranchWorkflow.java

```diff
@@ -0,0 +1,123 @@
+import jenkins.plugins.git.GitBranchSCMHead;
+import jenkins.plugins.git.GitSCMSource;
+import jenkins.plugins.git.GitSampleRepoRule;
+import jenkins.scm.api.SCMHead;
+import org.biouno.unochoice.AbstractUnoChoiceParameter;
+import org.biouno.unochoice.ChoiceParameter;
+import org.biouno.unochoice.model.GroovyScript;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SecureGroovyScript;
+import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
+import org.jenkinsci.plugins.scriptsecurity.scripts.languages.GroovyLanguage;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.*;
+import java.util.Arrays;
+
+import static org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProjectTest.scheduleAndFindBranchProject;
+import static org.junit.Assert.*;
+
+/**
+ * Test that scripts can access the jenkinsProject object.
+ *
+ * @since 2.6.5
+ */
+@Issue("JENKINS-65235")
+public class TestProjectObjectIsPresentOnMultibranchWorkflow {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+    @ClassRule public static BuildWatcher buildWatcher = new BuildWatcher();
+    @Rule public GitSampleRepoRule sampleRepo = new GitSampleRepoRule();
+
+    // LIST script
+    private final String SCRIPT_LIST = "return ['A', 'B', jenkinsProject.fullName, 'C']";
+    private final String FALLBACK_SCRIPT_LIST = "return ['EMPTY!']";
+
+    @Before
+    public void setUp() throws Exception {
+
+        ScriptApproval.get()
+                .preapprove(SCRIPT_LIST, GroovyLanguage.get());
+        ScriptApproval.get()
+                .preapprove(FALLBACK_SCRIPT_LIST, GroovyLanguage.get());
+    }
+
+    @Test
+    public void testProjectObjectIsAvailable() throws Exception {
+        // create a multibranch workflow from git repo
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Jenkins.ADMINISTER).everywhere().toEveryone()
+        );
+        // create a git repo
+        sampleRepo.init();
+        sampleRepo.write("Jenkinsfile", "echo 'it works'");
+        sampleRepo.git("add", "Jenkinsfile");
+        sampleRepo.git("commit", "--all", "--message=flow");
+        // create a multibranch workflow from git repo
+        WorkflowMultiBranchProject mp = j.createProject(WorkflowMultiBranchProject.class, "gitprj");
+        mp.getSourcesList().add(new BranchSource(new GitSCMSource(null, sampleRepo.toString(), "", "*", "", false)));
+        WorkflowJob gitprj = scheduleAndFindBranchProject(mp, "master");
+        assertEquals(new GitBranchSCMHead("master"), SCMHead.HeadByItem.findHead(gitprj));
+        assertEquals(1, mp.getItems().size());
+        j.waitUntilNoActivity();
```

Comment:
Didn't know about `waitUntilNoActivity`, tests are probably less flaky using this :+1: thanks!